oauth2_provider/models.py

```diff
@@ -434,8 +436,18 @@ def get_refresh_token_model():
     return apps.get_model(oauth2_settings.REFRESH_TOKEN_MODEL)
 
 
-def clear_expired():
-    now = timezone.now()
+def clear_expired(before = None):
+    """Clear expired tokens, add custom date to limit clear tokens before this date."""
+    if not before:
+        now = timezone.now()
+    else:
+        now = before
+        if not type(before).__name__ == 'datetime':
```

Comment:
Is this branch needed since management command uses `dateutil.parser.parse`?

Reply:
we need this, because celery task call this function direct, without management command

Reply:
Could Celery task call it with `datetime` rather than a `string`?

Reply:
celery call without extra params, but better is protect this function from incorrect input data

Reply:
Ok, you could check for the type `isinstance`, log if this is not a valid `datetime`and return